lib/execution_segment.go

```diff
@@ -293,3 +300,97 @@ func (es *ExecutionSegment) CopyScaleRat(value *big.Rat) *big.Rat {
 	}
 	return new(big.Rat).Mul(value, es.length)
 }
+
+// ExecutionSegmentSequence represents an ordered chain of execution segments,
+// where the end of one segment is the beginning of the next. It can serialized
+// as a comma-separated string of rational numbers "r1,r2,r3,...,rn", which
+// represents the sequence (r1, r2], (r2, r3], (r3, r4], ..., (r{n-1}, rn].
+// The empty value should be treated as if there is a single (0, 1] segment.
+type ExecutionSegmentSequence []*ExecutionSegment
+
+// NewExecutionSegmentSequence validates the that the supplied execution
+// segments are non-overlapping and without gaps. It will return a new execution
+// segment sequence if that is true, and an error if it's not.
+func NewExecutionSegmentSequence(segments ...*ExecutionSegment) (ExecutionSegmentSequence, error) {
+	if len(segments) > 2 {
```

Comment:
shouldn't this be `>=`?

Reply:
Resolved by https://github.com/loadimpact/k6/pull/1318/commits/686f32b9204b02f4214be812ad2ec1966b4463fc